src/ad_extension.js

```diff
@@ -1,16 +1,13 @@
-export class AdExtension {
-  constructor() {
-    this.name = null;
-    this.value = null;
-    this.attributes = {};
-    this.children = [];
-  }
-
-  isEmpty() {
+export function AdExtension() {
+  this.name = null;
```

Comment:
we should not assign the values to `this`, because then we need to depend on the context. We should return a new immutable object